Lib/pathlib/_abc.py

```diff
@@ -799,13 +806,8 @@ def glob(self, pattern, *, case_sensitive=None, follow_symlinks=None):
                 raise ValueError("Invalid pattern: '**' can only be an entire path component")
             else:
                 dir_only = bool(stack)
-                match = _compile_pattern(part, sep, case_sensitive)
+                match = _compile_pattern(part, sep, case_sensitive) if part != '*' else None
```

Comment:
```suggestion
                # Pass in a pattern to eagerly exclude paths we know don't match.
                # Otherwise, they'll end up in the final set and be filtered anyway.
                match = _compile_pattern(part, sep, case_sensitive) if part != '*' else None
```